File: q431/helloworld.hs
-- import Foreign.Storable
import Data.Char
-- one = length $ show $ sizeOf $ True -- 4
one = length $ return undefined -- https://gist.github.com/takeouchida/7669018
two = one+one
x n = two*n
-- y n = two*n+one
y n = succ $ x n
main = do
	putChar $ chr $ x $ x $ x $ y $ x $ x $ one
	putChar $ chr $ y $ x $ y $ x $ x $ y $ one
	putChar $ chr $ x $ x $ y $ y $ x $ y $ one
	putChar $ chr $ x $ x $ y $ y $ x $ y $ one
	putChar $ chr $ y $ y $ y $ y $ x $ y $ one
	putChar $ chr $ x $ x $ x $ x $ x $ one
	putChar $ chr $ y $ y $ y $ x $ y $ x $ one
	putChar $ chr $ y $ y $ y $ y $ x $ y $ one
	putChar $ chr $ x $ y $ x $ x $ y $ y $ one
	putChar $ chr $ x $ x $ y $ y $ x $ y $ one
	putChar $ chr $ x $ x $ y $ x $ x $ y $ one
	putChar $ chr $ x $ y $ x $ one

File: q431/helloworld.erl
-module(prog).
-export([main/0]).

one() -> size({true}).
two() -> one()*one()+one().
x(N) -> two()*N.
y(N) -> two()*N+one().

main() -> io:put_chars([
	x(x(x(y(x(x(one())))))),
	y(x(y(x(x(y(one())))))),
	x(x(y(y(x(y(one())))))),
	x(x(y(y(x(y(one())))))),
	y(y(y(y(x(y(one())))))),
	x(x(x(x(x(one()))))),
	y(y(y(x(y(x(one())))))),
	y(y(y(y(x(y(one())))))),
	x(y(x(x(y(y(one())))))),
	x(x(y(y(x(y(one())))))),
	x(x(y(x(x(y(one())))))),
	x(y(x(one())))
]).